### PR TITLE
feat(HOME-3012): MFA

### PR DESCRIPTION
**Pull request by antoinemeyer**
https://trizic.atlassian.net/browse/HOME-3012

### CODE REVIEW

File: api/v3/auth/login.raml [hunk 1]
```diff
@@ -39,5 +39,14 @@
...
             example: !include examples/login.res.json
       401: !include ../../common/unauthorized.raml
-
-
+      403:
+        description: |
+          Error message may be of the following:
+          -EMAIL_NOT_VERIFIED: User need to verify login before being able to login. 
+          -MFA_TOKEN_REQUIRED: MFA is required but token was not provided. 
+          -MFA_TOKEN_INVALID: MFA Token is invalid.
```
Comment (kvhswong) [inline api/v3/auth/login.raml]: Missing this enum per the Jira, MFA_CONFIGURATION_REQUIRED.

Comment (antoinemeyer) [reply api/v3/auth/login.raml]: That gets actually returned in a 200 https://github.com/trizic/api-docs-consumer/pull/180/files/e427bdaba2093689e868fe40d1f731b01db641e9#diff-a2889c4ac36c75e06c2b3aeb70a01847R19

File: api/v3/auth/index.raml [hunk 2]
```diff
@@ -1,2 +1,24 @@
 #%RAML 0.8
 /login: !include ./login.raml
+/login/sms: 
+  get:
```
Comment (kvhswong) [inline api/v3/auth/index.raml]: These were described as POSTs in the Jira. Since it changes state, prob want to keep it as a POST.

Comment (kvhswong) [reply api/v3/auth/index.raml]: Plus, I believe that the user has not authenticated at this point so credentials would be required.

File: api/v3/auth/index.raml [hunk 3]
```diff
@@ -1,2 +1,24 @@
...
+    body:
+      application/json:
+        example: !include examples/login.req.json
+        schema: !include schema/login.req.json
+    responses:
+      200:
+      401: !include ../../common/generic.raml
+      429: !include ../../common/generic.raml
+/login/call: 
+  get:
```
Comment (kvhswong) [inline api/v3/auth/index.raml]: Same comment here as above.

File: api/v3/auth/setup/index.raml [hunk 4]
```diff
@@ -0,0 +1,58 @@
+#%RAML 0.8
+/:
```
Comment (kvhswong) [inline api/v3/auth/setup/index.raml]: Don't think this "/: is required as it's showing up with a slash appended to the path segment.

screen shot 2018-02-14 at 4 38 46 pm

File: api/v3/auth/setup/index.raml [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+      403:
+        description: |
+          Error message may be of the following:
+          -MFA_TOKEN_INVALID: MFA Token is invalid.
+        body:
+          application/json:
+            schema: !include ../../../common/schema/generic.res.json
+            example: !include ../../../common/examples/generic.res.json
+/sms:
+  get:
```
Comment (kvhswong) [inline api/v3/auth/setup/index.raml]: Could argue if these three routes are really GET's, but no need to be pedantic.

**Comment by antoinemeyer** [comment]
@kevinbrosamle looks good to you?

**Pull request merged**